File: lab5/TVector2D.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
    public class TVector2D
    {
    protected double x;
    protected double y;

    public TVector2D(double x, double y)
    {
        this.x = x;
        this.y = y;
    }
    public TVector2D()
    {
        this.x = 0;
        this.y = 0;
    }

    public TVector2D( TVector2D vector)
    {
        this.x = vector.x;
        this.y = vector.y;
    }

    public virtual void input()
    {
        Console.WriteLine("Please, enter x");
        x=Int32.Parse(Console.ReadLine());
        Console.WriteLine("Please, enter y");
        y =Int32.Parse(Console.ReadLine());
    }

    public virtual void print()
    {
        Console.Write(x + "; " + y + " ");
    }

    public virtual double length()
    {
        return (Math.Sqrt(x * x + y * y));
    }

    public virtual void normalize()
    {
        double length = this.length();
        x = x / length;
        y = y / length;

    }
    //ToDo чи треба додавати протилежність?
    public bool equals(TVector2D vector) 
    {
        return x == vector.x && y == vector.y;
    }

    public static TVector2D operator + (TVector2D vector1, TVector2D vector2)
    {
        return new TVector2D(vector1.x + vector2.x, vector1.y + vector2.y);
    }
    public static TVector2D operator -(TVector2D vector1, TVector2D vector2)
    {
        return new TVector2D(vector1.x - vector2.x, vector1.y - vector2.y);
    }
    public static double operator *(TVector2D vector1, TVector2D vector2)
    {
        return (vector1.x * vector2.x + vector1.y * vector2.y);
    }

}

File: lab5c_plus_plus/TVector2D.cpp
#include<iostream>;
#include<Math.h>
using namespace std;
class TVector2D
{
protected:
    double x;
    double y;

public:
    TVector2D(double x, double y)
    {
        this->x =  x;
        this->y =  y;
    }
    TVector2D()
    {
        x = 0;
        y = 0;
    }

    TVector2D(const TVector2D &vector)
    {
        x = vector.x;
        y = vector.y;
    }

    virtual void input()
    {
        cout<<"Please, enter x"<<endl;
        cin >> x;
        cout<<"Please, enter y"<<endl;
        cin >> y;
    }

    void virtual print()
    {
       cout<<x << "; " << y << " ";
    }

    double virtual length()
    {
        return (sqrt(x * x + y * y));
    }

    void virtual normalize()
    {
        double lenght = this->length();
        x = x / lenght ;
        y = y / lenght;

    }
    //ToDo чи треба додавати протилежність?
    bool equals(TVector2D vector)
    {
        return x == vector.x && y == vector.y;
    }

     TVector2D operator + (const TVector2D vector2)
    {
        return TVector2D(this->x + vector2.x, this->y + vector2.y);
    }
     TVector2D operator -(const TVector2D vector2)
    {
        return TVector2D(this->x - vector2.x, this->x - vector2.y);
    }
     double operator *(const TVector2D vector2)
    {
        return (this->x * vector2.x + this->x * vector2.y);
    }

};